utils/src/main/java/software/amazon/awssdk/utils/http/SdkHttpUtils.java

```diff
@@ -335,10 +332,26 @@ public static boolean isSingleHeader(String h) {
      * Extracts query parameters from the given URI
      */
     public static Map<String, List<String>> uriParams(URI uri) {
-        return QUERY_PARAM_DELIMITER_PATTERN
-                      .splitAsStream(uri.getRawQuery().trim())
-                      .map(s -> s.contains("=") ? s.split("=", 2) : new String[] {s, null})
-                      .collect(groupingBy(a -> urlDecode(a[0]), mapping(a -> urlDecode(a[1]), toList())));
+        return splitQueryString(uri.getRawQuery())
+            .stream()
+            .map(s -> s.contains("=") ? s.split("=", 2) : new String[] {s, null})
```

Comment:
Is `contains` necessary? Can we just uniformly call `split`?

Reply:
Good question, I just copied+pasted that part. 

split("=", 2) will only return 1 element if there is no =, so we still need to account for that case, but it's a little more efficient to do it after the split than before. Will update the PR.